tianshou/env/vecenv/shmemenv.py

```diff
@@ -0,0 +1,175 @@
+import gym
+import ctypes
+import numpy as np
+from collections import OrderedDict
+from multiprocessing import Pipe, Process, Array
+from typing import Callable, List, Optional, Tuple, Union
+
+
+from tianshou.env import BaseVectorEnv, SubprocVectorEnv
+from tianshou.env.utils import CloudpickleWrapper
+
+_NP_TO_CT = {np.bool: ctypes.c_bool,
+             np.uint8: ctypes.c_uint8,
+             np.uint16: ctypes.c_uint16,
+             np.uint32: ctypes.c_uint32,
+             np.uint64: ctypes.c_uint64,
+             np.int8: ctypes.c_int8,
+             np.int16: ctypes.c_int16,
+             np.int32: ctypes.c_int32,
+             np.int64: ctypes.c_int64,
+             np.float32: ctypes.c_float,
+             np.float64: ctypes.c_double}
```

Comment:
How about np.int, np.bool_, and np.float?

Reply:
np.int == int and is not supported by gym(np.float the same)